samples/host_exerciser/host_exerciser_cmd.h

```diff
@@ -117,11 +151,41 @@ class host_exerciser_cmd : public test_command
 
     virtual int run(test_afu *afu, CLI::App *app)
     {
+        fpga_result res=FPGA_OK;
         (void)app;
 
         auto d_afu = dynamic_cast<host_exerciser*>(afu);
         host_exe_ = dynamic_cast<host_exerciser*>(afu);
 
+        token_ = d_afu->get_token();
+
+        //fpga perf counter initialization
+        fpgaperf fpgaperf;
+
+        fpga_perf_counter *fpga_perf = new fpga_perf_counter;
+
+        res = fpgaperf.createhandle(fpga_perf);
+        if (res != FPGA_OK) {
+                return -1;
+        }
+
+        res = fpgaperf.perfget(token_, fpga_perf);
+        if (res != FPGA_OK) {
+                fpgaperf.perfdestroy(fpga_perf);
+                fpgaperf.destroyhandle(fpga_perf);
+                delete fpga_perf;
+                fpga_perf = nullptr;
```

Comment:
I'm seeing a pattern here when we want to abort and have to cleanup the fpga_perf structure.
Perhaps this could be refactored/wrapped by a function or a c++ object that does lifecycle management?